### PR TITLE
Fixed possible indentation error in the training loop in utils/utils.py.

### PR DESCRIPTION
**Pull request by JLA0809**
I think there could be an indendation error in the main training loop in the utils/utils.py in the original main branch. In my opinion the optimizer has to be called for every batch, in order to use the information of the whole training dataset.

### CODE REVIEW

**Review by XinyiYS — CHANGES_REQUESTED** [review]
I think you are correct. Would you please indent the line no. 134 where the scheduler also takes a step? I will approve the PR afterwards.

**Comment by JLA0809** [comment]
Sorry for the delay, the line should now be indented in the latest commit
